libkrun/test/test_libkruntime.py

```diff
@@ -83,6 +83,21 @@ def test_cycles_u64_double_ratio(self):
     def test_clock_gettime_monotonic(self):
         rv, out, _ = invoke_c_prog("clock_gettime_monotonic")
         assert rv == 0
-        dct = parse_keyvals(out)
+        dct = parse_keyvals(out, True)
         # Depends on speed of CPU, but should be very close to 1
         assert 0.95 <= dct["monotonic_delta"] <= 1.05
+
+    def test_msr_time(self):
+        rv, out, _ = invoke_c_prog("msr_time")
+        assert rv == 0
+        dct = parse_keyvals(out)
+
+        # I expect reading MSRs to be more than 70x slower than reading MSRs.
```

Comment:
I think this slowdown is purely because of Linux's file-system mechanism for reading MSRs. I don't think it *has* to be that slow! I think we should drop the 70x thing (which is pretty arbitrary, and may or may not hold on other machines), and just assert that the MSRs are slower than the monotonic timer. We may also want to make this test Linux only? I doubt reading MSRs will be as slow on OpenBSD. Perhaps we can address that later, though.

So we might want to say "Reading an MSR on Linux is *much* slower than reading wall-clock time. If this were to not be the case, we might want to reorder the way that time/MSRs are read."